lib/recipe.rb

```diff
@@ -55,13 +66,46 @@
         'entity_uuid'        => Chef::JSONCompat.parse(Chef::FileCache.load('data_collector_metadata.json'))['node_uuid'],
         'install_check_file' => Gem.ruby,
         'org_name'           => Chef::Config[:data_collector][:organization] || server_uri.path.split('/').last,
-        'unprivileged_uid'   => nil,
-        'unprivileged_gid'   => nil,
+        'unprivileged_uid'   => agent_user.uid,
+        'unprivileged_gid'   => agent_user.gid,
+        'scheduled_task_mode' => platform?('windows')
       }.to_json
     end
   )
 end
 
+if platform?('windows')
+
+  # In windows we run things as as a scheduled task instead of a daemon
+  # This avoids the code overhead of a service, and isn't too inefficient since we run infrequently
+  scheduled_task_script = ChefConfig::PathHelper.join(agent_bin_dir, "automate_liveness_agent_task.ps1")
+
+  # We hide configuration details in this script; it was too much to pass on the schtasks command line
+  file scheduled_task_script do
+    mode 0755
+    owner install_user
+    group install_group
+    content <<"SCRIPT_BODY"
+# powershell script to run the Chef automate liveness agent as a scheduled task\r\n
+$env:RUBYOPT = "--disable-gems"\r\n
+$env:RUBY_GC_HEAP_GROWTH_MAX_SLOTS = 500\r\n
+#{Gem.ruby} #{agent_bin} #{agent_conf}\r\n
+SCRIPT_BODY
+    # debugging add Get-Date -Format g | Out-File c:\\chef\\script.log\r\n to script above
+  end
+
+  # Set up scheduled task; this is idempotent because scheduled tasks replace ones with same name
```

Comment:
Some folks might be grumpy to see their chef run have more than zero updated resources since they kinda fixate on that as their measurement of idempotence. We can make a ticket to fix that later if necessary.